kyaml/kio/pkgio_reader_test.go

```diff
@@ -555,3 +562,41 @@ func testOnDiskAndOnMem(t *testing.T, files []mockFile, f func(t *testing.T, pat
 		f(t, "/", fs)
 	})
 }
+
+func TestLocalPackageReader_ReadBareSeqNodes(t *testing.T) {
+	testOnDiskAndOnMem(t, []mockFile{
+		{path: "a/b"},
+		{path: "a/c"},
+		{path: "e_test.yaml", content: readFileE},
+	}, func(t *testing.T, path string, mockFS filesys.FileSystem) {
+		rfr := LocalPackageReader{
+			PackagePath:     path,
+			FileSystem:      filesys.FileSystemOrOnDisk{FileSystem: mockFS},
+			WrapBareSeqNode: true,
+		}
+		nodes, err := rfr.Read()
+		require.NoError(t, err)
+		require.Len(t, nodes, 2)
+		expected := []string{
+			`a: b #first
+metadata:
+  annotations:
+    config.kubernetes.io/index: '0'
+    config.kubernetes.io/path: 'e_test.yaml'
+`,
+			`bareSeqNodeWrappingKey:
+- foo # second
+- bar
+metadata:
+  annotations:
```

Comment:
Considering that this is not valid yaml (running it though yaml linters throws an error), I'm wondering if this is really the best option we have. Are there any alternative ways to support bare sequence nodes without having to generate and support invalid yaml?

Reply:
It is a valid YAML.

Reply:
Are you certain? Every yaml linter I run it through (python and several others found online) state otherwise.

Reply:
did you include line 587 ?

Reply:
Oh my mistake, sorry I didn't see that. Thanks for catching that.